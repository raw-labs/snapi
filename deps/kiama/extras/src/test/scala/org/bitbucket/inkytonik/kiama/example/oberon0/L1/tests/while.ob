MODULE while;

VAR
    avar : INTEGER;
    bvar : BOOLEAN;

BEGIN
    WHILE bvar DO
    END;
    
    WHILE bvar DO
        avar := 1
    END;
    
    WHILE bvar DO
        avar := 2;
    END;

    WHILE (avar > 1) & (avar > 0) DO
        avar := avar + 4;
        bvar := avar < 5
    END
END while.
