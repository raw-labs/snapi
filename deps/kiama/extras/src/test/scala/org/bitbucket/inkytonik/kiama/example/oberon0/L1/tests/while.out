* _pp.ob
MODULE while;

    VAR
        avar : INTEGER;
        bvar : BOOLEAN;

BEGIN
    WHILE bvar DO
        
    END;
    WHILE bvar DO
        avar := 1
    END;
    WHILE bvar DO
        avar := 2;
        
    END;
    WHILE (avar > 1) & (avar > 0) DO
        avar := avar + 4;
        bvar := avar < 5
    END
END while.
[6]* c
[6]#include <stdio.h>
[6]int ob_avar2;
[6]int ob_bvar3;
[6]int main (int argc, char *argv[0]) {
[6]    while (ob_bvar3) {
[6]        ;
[6]    }
[6]    while (ob_bvar3) {
[6]        ob_avar2 = 1;
[6]    }
[6]    while (ob_bvar3) {
[6]        ob_avar2 = 2;
[6]        ;
[6]    }
[6]    while (ob_avar2 > 1 && ob_avar2 > 0) {
[6]        ob_avar2 = ob_avar2 + 4;
[6]        ob_bvar3 = ob_avar2 < 5;
[6]    }
[6]    return 0;
[6]}
